docs/portal/developer-portal/advanced/Enabling_K3s.md

```diff
@@ -0,0 +1,317 @@
+
+There are alternate configurations of podman that would allow for different workflows, for example, the main pid of the container could be long running, to facilitate easier reentry to the container on subsequent logins. 
+
+## Prerequisites 
+
+The following sections should be completed prior to configuring the UAN with K3s.
+
+### UAN K3s Node
+Designate a UAN to operate as the K3s control-plane node.
+
+**Note**: In a future release, additional UANs will be able to join as extra manager or worker nodes. 
+
+### MetalLB IP Address Pool
+
+A pool of IP Address should be configured for services running in K3s. This will allow for external `LoadBalancer` IP Address to be assigned to services like `HAProxy`.  This address pool must be routable from the UAN control-plane, and should be unused for other purposes. Initially, these IP address will serve as the SSH ingress for instances of `HAProxy`.
+
+### Subuid and Subgid
+
+To allow for users to run rootless podman containers, the files `/etc/subuid` and `/etc/subgid` must be present and configured with an entry for each user. These files should be uploaded to the `user` S3 bucket:
+```bash
+$ cray artifacts list user --format json
+{
+  "artifacts": [
+    {
+      "Key": "subuid",
+      "LastModified": "2023-02-21T23:41:43.948000+00:00",
+      "ETag": "\"c543aebb9b40bcf48879885734447090\"",
+      "Size": 145686,
+      "StorageClass": "STANDARD",
+      "Owner": {
+        "DisplayName": "User Service User",
+        "ID": "USER"
+      }
+    },
+    {
+      "Key": "subgid",
+      "LastModified": "2023-02-21T23:41:43.948000+00:00",
+      "ETag": "\"73032ede132e44d2c1bc567246901737\"",
+      "Size": 145686,
+      "StorageClass": "STANDARD",
+      "Owner": {
+        "DisplayName": "User Service User",
+        "ID": "USER"
+      }
+    }
+  ]
+}
+```
+
+### Podman Image
+A container image suitable for users should be available in a container image registry accessible from the UAN.
+
+
+## Configuring with Configuration Framework Service (CFS)
+
+After completing the prerequisites section, the following should be established to proceed with configuring a UAN to run K3s.
+
+- A fully configured UAN
+- An IPAddress start and end range to assign to MetalLB
+- Prepared subuid and subgid files 
+
+### Configuration Files and Playbook
+
+Configuration for K3s and related services are controlled in the following ansible files in the UAN VCS repository:
+```bash
+$ ls uan-config-management/vars/ | egrep "k3s|sshd|helm"
+uan_helm.yml
+uan_k3s.yml
+uan_sshd.yml
+```
+The ansible playbook to install and configure K3s may be found here:
+```bash
+$ ls uan-config-management | grep k3s
+k3s.yml
+```
+
+### Ansible Roles
+
+The following Ansible roles are provided in the `uan-config-management` repository in VCS. There are `README.md` files in each Ansible role directory in this repository that provide further details.
```

Comment:
Should we update the "Ansible Roles" file in the documentation now for these new roles, or wait until this is more of a stable, supported feature?

Reply:
I would prefer to wait. Also, I'd like to come up with a way to extract those READMEs programmatically instead of duplicating the information as it inevitably becomes out of sync.